src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCaller.java

```diff
@@ -190,10 +193,14 @@ public AssemblyRegionEvaluator assemblyRegionEvaluator() {
 
     @Override
     public void onTraversalStart() {
+        if(hcArgs.genotypeArgs.applyPriors && hcArgs.genotypeArgs.samplePloidy != HomoSapiensConstants.DEFAULT_PLOIDY) {
+            throw new UserException("Priors can only be applied for diploid samples.");
```

Comment:
This check should be moved to `HaplotypeCallerEngine.validateAndInitializeArgs()`

Reply:
Is this moveable? It's gross having it in the tool itself rather than the `HaplotypeCallerEngine`